contracts/InterchainTokenFactory.sol

```diff
@@ -404,10 +404,24 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
         bytes32 deploySalt = canonicalInterchainTokenDeploySalt(tokenAddress);
         string memory currentChain = '';
         uint256 gasValue = 0;
+        _checkToken(tokenAddress);
 
         tokenId = interchainTokenService.deployTokenManager(deploySalt, currentChain, TokenManagerType.LOCK_UNLOCK, params, gasValue);
     }
 
+    function _checkToken(address tokenAddress) internal view {
```

Comment:
create a helper `_getTokenMetadata` that returns name, symbol, decimals and has a single try catch that reverts with NotToken if this fails. This helper can then be used in other functions in this contract for consistency, since the token could be upgraded between deployment calls

Reply:
The reason why that's not optimal is because putting `name` and `symbol` in memory sill use gas even if the values are not used after. Also try-catch are only available in external calls, so to do what you are suggesting I would need to make the call external, which is even more gas, and more gas for other uses as well. Given the above, I think this would be a bad idea, lmk if you wanna do it anyway.

Reply:
Since it's a deployment method, a bit more gas is fine to make this more consistent whenever metadata is retrieved across multiple functions. You can keep the 3 try-catches since it can't be used on an internal method

Reply:
So are you suggesting we do it in an internal method with the try-catches then? It is not very clear to me.